photon.c

```diff
@@ -0,0 +1,135 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <inttypes.h>
+
+#include "utarray.h"
+#include "common.h"
+#include "state/task_queue.h"
+#include "state/db.h"
+#include "event_loop.h"
+#include "task.h"
+#include "photon.h"
+#include "io.h"
+
+typedef struct {
+  event_loop *loop;
```

Comment:
Do we need a pointer to the event loop in the local scheduler state? It looked like it was being passed around as part of the event loop processing, but I'm not sure.